Src/Web/Web.Shared.Net/SyntheticUserAgentTelemetryInitializer.cs

```diff
@@ -49,19 +51,26 @@ protected override void OnInitializeTelemetry(HttpContext platformContext, Reque
             {
                 if (platformContext != null)
                 {
-                    var request = platformContext.GetRequest();
-
-                    if (request != null && !string.IsNullOrEmpty(request.UserAgent))
+                    if (platformContext.Items.Contains(SyntheticSourceNameKey))
```

Comment:
I agree that code looks simpler without that check, but from scenario perspective it will be better to store it in requesttelemetry